### PR TITLE
Expands todo data into v2 format

### PR DESCRIPTION
**Pull request by scalvert**
The v2 format is necessary to enable the new fuzzy matching feature in #252. 

These changes include

- [x] On read, normalizing the format to v2
- [x] On write, writing v2 format
- [x] On operation and comparison, use v2 format

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
     "docs:update": "yarn build && node scripts/generate-api-docs.js",
     "lint": "eslint . --ext .ts",
     "prepare": "yarn build",
-    "test": "jest --no-cache"
+    "test": "yarn lint && yarn test:jest",
```
Comment (rwjblue) [inline package.json:16]: I'd _vaguely_ prefer something like what we do in Ember apps:

```
"test": "npm-run-all lint test:*",
```

File: src/builders.ts [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  Location,
+  Range,
+  TodoConfig,
+  TodoData,
+  TodoDataV1,
+  TodoDataV2,
+} from './types';
 import { getDatePart } from './date-utils';
 
+const LINES_PATTERN = /(.*?(?:\r\n?|\n|$))/gm;
```
Comment (rwjblue) [inline src/builders.ts:17]: Is this just for eslint? We should make an issue (so we don't forget) to extract this and force it into a consumer problem

File: src/todo-matcher.ts [hunk 3]
```diff
@@ -1,40 +1,44 @@
...
         todoDataToFind.engine === todoDatum.engine &&
         todoDataToFind.ruleId === todoDatum.ruleId &&
-        todoDataToFind.line === todoDatum.line &&
-        todoDataToFind.column === todoDatum.column
+        todoDataToFind.range.start.line === todoDatum.range.start.line &&
+        todoDataToFind.range.start.column === todoDatum.range.start.column &&
+        ((todoDataToFind.source &&
+          todoDatum.source &&
+          todoDataToFind.source === todoDatum.source) ||
+          true)
```
Comment (rwjblue) [inline src/todo-matcher.ts:41]: Can you split this into a separate variable? It's kinda difficult to parse/understand...

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: src/builders.ts (hunk 2 above)
Comment (scalvert) [reply src/builders.ts:17]: Yep agreed. I think we ultimately want to have consumers pass us pre-normalized v2 format. I'll create an issue and follow that up in a separate PR.

Comment (scalvert) [reply src/builders.ts:17]: Logged #258 to capture this requirement.

File: src/todo-matcher.ts (hunk 3 above)
Comment (scalvert) [reply src/todo-matcher.ts:41]: Agree, it's insane mental gymnastics right now.

File: package.json (hunk 1 above)
Comment (scalvert) [reply package.json]: Done.

File: src/todo-matcher.ts (hunk 3 above)
Comment (scalvert) [reply src/todo-matcher.ts]: Updated.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**